### DISCUSSION TITLE
Stream Plugin - Networking Issues

### DISCUSSION THREAD

**Discussion by nlm4145** [rabbitmq/rabbitmq-server#11313] [Questions]
Hello RabbitMQ community. I'm facing some connectivity issues with the stream plugin. Here's some context:

We have deployed up Rabbit MQ on a pod in our EKS Cluster but are finding difficulties maintaining between pod connectivity (between RabbitMQ and other applications) and maintaining port-forwarding capabilities from local machines.

If we want to let our other pods connect to RabbitMQ, the plugin configuration settings have to be:

```
  stream.advertised_host = rabbitmq-test.rabbitmq.svc.cluster.local # same as dnsConfig
  stream.advertised_port = 5552
```

But when port forwarding to the instance via:

` kubectl port-forward -n rabbitmq svc/rabbitmq-test 15672:15672 5672:5672 5552:5552`

And attempting to connect using a library, like rstream, we are not able to connect successfully.

we have to set the configs like this in order to connect locally via port-fowarding:

```
  stream.advertised_host = **localhost**
  stream.advertised_port = 5552
```

But then, our other pods are unable to connect to it, using localhost or rabbitmq-test.rabbitmq.svc.cluster.local.

How can we workaround this?

**Comment by lukebakken** [comment]
> But when port forwarding to the instance via:
> kubectl port-forward -n rabbitmq svc/rabbitmq-test 15672:15672 5672:5672 5552:5552
> And attempting to connect using a library, like rstream, we are not able to connect successfully

Questions:
* What hostname is your application using to connect? I'm assuming it's using `rabbitmq-test.rabbitmq.svc.cluster.local`.
* Why do you expect the combination of using that hostname, and port forwarding, to work?

> But then, our other pods are unable to connect to it, using localhost or rabbitmq-test.rabbitmq.svc.cluster.local.

Are you configuring port forwarding on your other pods?

**Reply by nlm4145** [reply]
- Yes, that is the hostname we our pods are attempting to connect to.
- We expect for our Pods to be able to connect via the host name referenced the point above but when port forwarding locally via localhost. It works fine when interacting with port 5672 regardless of the stream.advertised_host which makes me think this is stream plugin specific.

We haven't attempted to set up port forwarding on our local pods (i'm assuming you mean when stream.advertised_host is set as localhost - which is what most of the examples show using). Is there a recommended way?

**Reply by lukebakken** [reply]
https://www.rabbitmq.com/docs/stream#advertised-host-port

https://www.rabbitmq.com/blog/2021/07/23/connecting-to-streams
